src/components/PurchaseBook.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {useState, useContext} from 'react';
+import {Col, Row, Typography, Select, Button, Divider, message} from 'antd';
+import {Book} from '../models/Book';
+import { BasketContext } from '../state/basket/BasketContext';
+
+const { Option } = Select;
+const {Title, Text} = Typography;
+
+interface PurchaseProps{
+    book: Book
+}
+
+const PurchaseBook = ({book}: PurchaseProps) => {
+    const {basket, addToBasket} = useContext(BasketContext);
+    const isInStock = book.stockAmount > 0;
+
+    const maxQuantity = book.stockAmount < 20 ? book.stockAmount : 20;
```

Comment:
Don't like magic numbers. Pull that out into a const about and explain the 20 is.